docs/src/examples/jump_umap_analysis_with_cosmicqc.py

```diff
@@ -0,0 +1,421 @@
+            "Metadata_Well",
+        ],
+    )
+
+    df_feature_selected_with_cqc_outlier_data.to_parquet(parquet_sampled_with_outliers)
+
+else:
+    df_feature_selected_with_cqc_outlier_data = pd.read_parquet(
+        parquet_sampled_with_outliers
+    )
+
+df_feature_selected_with_cqc_outlier_data
+# -
+
+# show our data value counts regarding outliers vs inliers
+df_feature_selected_with_cqc_outlier_data[
+    "analysis.included_at_least_one_outlier"
+].value_counts()
+
+# prepare data for normalization and feature selection
+# by removing cosmicqc and analaysis focused columns.
+df_for_normalize_and_feature_select = df_feature_selected_with_cqc_outlier_data[
+    [
+        col
+        for col in df_feature_selected_with_cqc_outlier_data.columns.tolist()
+        if "cqc." not in col and "analysis." not in col
+    ]
+]
+# show the modified column count
+len(df_for_normalize_and_feature_select.columns)
+
+# +
+parquet_pycytominer_normalized = "./BR00117012_normalized.parquet"
+
+# check if we already have normalized data
+if not pathlib.Path(parquet_pycytominer_normalized).is_file():
+    # normalize the data using pcytominer
+    df_pycytominer_normalized = pycytominer.normalize(
+        profiles=df_for_normalize_and_feature_select,
+        features="infer",
+        image_features=False,
+        meta_features="infer",
+        method="standardize",
+        output_file=parquet_pycytominer_normalized,
+        output_type="parquet",
+    )
+
+# +
+parquet_pycytominer_feature_selected = "./BR00117012_feature_select.parquet"
+
+# check if we already have feature selected data
+if not pathlib.Path(parquet_pycytominer_feature_selected).is_file():
+    # feature select normalized data using pycytominer
+    df_pycytominer_feature_selected = pycytominer.feature_select(
+        profiles=parquet_pycytominer_normalized,
+        operation=[
+            "variance_threshold",
+            "correlation_threshold",
+            "blocklist",
+            "drop_na_columns",
+        ],
+        na_cutoff=0,
+        output_file=parquet_pycytominer_feature_selected,
+        output_type="parquet",
+    )
+
+
+# +
+def generate_umap_embeddings(
+    df_input: pd.DataFrame, cols_metadata: List[str]
+) -> np.ndarray:
+    """
+    Generates UMAP (Uniform Manifold Approximation and Projection)
+    embeddings for a given input dataframe,
+    excluding specified metadata columns.
+
+    Args:
+        df_input (pd.DataFrame]):
+            A dataframe which is expected to contain
+            numeric columns to be used for UMAP fitting.
+        cols_metadata (List[str]):
+            A list of column names representing
+            metadata columns that should be excluded
+            from the UMAP transformation.
+
+    Returns:
+        np.ndarray:
+            A dataframe containing the UMAP embeddings
+            with 2 components for each row in the input.
+    """
+
+    # Set constants
+    umap_n_components = 2
```

Comment:
prob should be a default arg?

Reply:
Agreed. Also maybe have a random state set here too unless I missed it.

Reply:
Ah good point! Thanks, will add in.

Reply:
I initially left random state out of the UMAP configuration to help with speed of processing (setting random state disallows parallel processing). I've added parameter with a default of "None" for this to the function for now, but wanted to double check: should we stick with setting a random state here? @jenna-tomkinson - thoughts?

Reply:
Will stick with a static random state here.